src/wallet/wallet2.cpp

```
@@ -4140,8 +4345,75 @@ void wallet2::setup_keys(const epee::wipeable_string &password)
+    {
+      THROW_WALLET_EXCEPTION_IF(!background_cache_password, error::wallet_internal_error,
+          "expected custom background cache password");
+      break;
+    }
+    default: THROW_WALLET_EXCEPTION(error::wallet_internal_error, "unknown background sync type");
+  }
+}
+//----------------------------------------------------------------------------------------------------
+void get_custom_background_key(const epee::wipeable_string &password, crypto::chacha_key &custom_background_key, const uint64_t kdf_rounds)
+{
+  crypto::chacha_key key;
```

Clang (14.0.6)

[warning] variable 'key' is not initialized

(cppcoreguidelines-init-variables)